cmd/adsysd/daemon/daemon_test.go

```diff
@@ -292,8 +292,12 @@ sssd:
 		filepath.Join(dir, "run"),
 		serviceTimeout))
 
-	err := os.WriteFile(configFile, data, 0600)
+	f, err := os.OpenFile(configFile, os.O_RDWR|os.O_CREATE, os.ModePerm)
+	require.NoError(t, err, "Setup: failed to open test config file")
+	defer f.Close()
+	_, err = f.Write(data)
 	require.NoError(t, err, "Setup: failed to write test config file")
+	f.Close()
```

Comment:
I thought about it and it seems like a good idea, tbh. This way we can prevent this amount of repetition.

Reply:
Do you mind commenting why we need to do this (I think you did add the comment in aad-auth), so that we don’t simplify it by error (especially as the test still pass after the simplification, so we won’t notice immediatly). Basically, your commit message is helpful.

Reply:
I'll do a better commenting on it after we extract it to `testutils`, if that's ok. It's better to do it in a different PR, right?

Reply:
as you are adding that replacement here, I think you can factor this directly in testutils and replace the WriteAll() calls by it

Reply:
Ack, will do!